tests/unit/ui/components/TestShowMoreCard.js

```diff
@@ -1,105 +1,223 @@
+    const root = render({
+      store,
+      children: 'Some text',
+    });
 
-    sinon.assert.calledWith(truncateToMaxHeight, contentNode);
+    root.setProps({
+      children: 'Some new text',
+    });
+
+    sinon.assert.called(dispatchSpy);
+  });
+
+  it('does not dispatch if the html children is the same', () => {
+    const { store } = dispatchClientMetadata();
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    /* eslint-disable react/no-danger */
+    const root = render({
+      store,
+      children: (
+        <div
+          dangerouslySetInnerHTML={{
+            __html: '<span>Some component text.</span>',
+          }}
+        />
+      ),
+    });
+
+    root.setProps({
+      children: (
+        <div
+          dangerouslySetInnerHTML={{
+            __html: '<span>Some component text.</span>',
+          }}
+        />
+      ),
+    });
+    /* eslint-enable react/no-danger */
+
+    sinon.assert.notCalled(dispatchSpy);
   });
 
-  it('expands if new child content is smaller', () => {
-    const root = mount(
-      <ShowMoreCardBase i18n={fakeI18n()}>
-        This would be very long content, but we cannot get `clientHeight` in the
-        tests, so this will be forced below (via setState()).
-      </ShowMoreCardBase>,
+  it('does not dispatch if the children is the same', () => {
+    const { store } = dispatchClientMetadata();
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    const root = render({
+      store,
+      children: 'Some text',
+    });
+
+    root.setProps({
+      children: 'Some text',
+    });
+
+    sinon.assert.notCalled(dispatchSpy);
+  });
+
+  it('executes truncateToMaxHeight when it receives props', () => {
+    const root = render();
+
+    const contentNode = root.instance().contents;
+
+    const truncateToMaxHeight = sinon.spy(
+      root.instance(),
+      'truncateToMaxHeight',
     );
 
-    // We have to manually set the expanded flag to false because we don't have
-    // a `clientHeight` in the tests.
-    root.setState({ expanded: false });
+    // We are simulating the truncate method call.
+    root.instance().truncateToMaxHeight({ clientHeight: MAX_HEIGHT + 1 });
+
+    // We are simulating any kind of update to properties.
+    root.setProps();
+
+    sinon.assert.calledWith(truncateToMaxHeight, contentNode);
+  });
+
+  it('does not execute truncateToMaxHeight when "read more" has been expanded', () => {
+    const { store } = dispatchClientMetadata();
 
-    expect(root.state('expanded')).toEqual(false);
-    // This will call `componentWillReceiveProps()`, the content of `children`
-    // is for example purpose.
-    root.setProps({ children: 'short content' });
-    expect(root.state('expanded')).toEqual(true);
+    // Simulating read more has been expanded already.
+    const root = render({
+      store: {
+        ...store,
+        store: {
+          uiState: {
+            readMoreExpanded: true,
+          },
```

Comment:
I am hoping this is ok or do I need to show it all out:

it hits truncate
it shows read more
user clicks on read more
state is change to readMoreExpanded: true
this calls componentWillReceive Props again and now it should not update. 

I wasn't sure...I kind of skipped to the part where is was just true. Let me know if that's a problem..

Reply:
I think because you're specifically testing that it doesn't call `truncateToMaxHeight` if it's already expanded, then just starting the test in the expanded state is fine.